### PR TITLE
bpo-32318: Don't call 'globals()' in socket.accept()

### PR DESCRIPTION
**Pull request by 1st1**
https://bugs.python.org/issue32318

### CODE REVIEW

**Comment by JulienPalard** [comment]
LGTM

File: Lib/socket.py [hunk 1]
```diff
@@ -203,10 +205,12 @@ def accept(self):
         For IP sockets, the address info is a pair (hostaddr, port).
         """
         fd, addr = self._accept()
-        # If our type has the SOCK_NONBLOCK flag, we shouldn't pass it onto the
-        # new socket. We do not currently allow passing SOCK_NONBLOCK to
-        # accept4, so the returned socket is always blocking.
-        type = self.type & ~globals().get("SOCK_NONBLOCK", 0)
+        type = self.type
+        if _SOCK_NONBLOCK:
```
Comment (asvetlov) [inline Lib/socket.py:209]: Either `_SOCK_NONBLOCK` should be `None` if not present in `socket` -- then `if _SOCK_NONBLOCK is not None:` will be very fast check.
Or drop `if`: `type & ~0` works pretty well

Comment (vstinner) [reply Lib/socket.py:209]: I agree with Andrew, I don't think that the if is needed.

**Comment by asvetlov** [comment]
Should we close the issue?
`socket.type` is fixed, isn't it?

**Comment by 1st1** [comment]
Yes, closing this one.

**Pull request closed without merging**